harvester_adapters/ckan/api.py

```diff
@@ -294,10 +294,14 @@ def create_package(self, ckan_package,
             # another posible [error] = {'owner_org': ['Organization does not exist']}
 
             # Check for duplicates
-            if json_content['error'].get('name', None) == ["That URL is already in use."]:
-                logger.error(f'Package Already exists! ACTION: {on_duplicated}')
+            dataset_exists = "That URL is already in use" in json_content['error'].get('name', '')
+            harvest_exists = "There already is a Harvest Source for this URL" in json_content['error'].get('url', '')
+            is_duplicated = dataset_exists or harvest_exists
+            if is_duplicated:
+                logger.error(f'Already exists! ACTION: {on_duplicated}')
                 if on_duplicated == 'SKIP':
-                    return {'success': True}
+                    # returns {'success': True, 'result': {the package}}
+                    return self.show_package(ckan_package_id_or_name=ckan_package['name'])
```

Comment:
does this really return {'success': True, 'result': {the package}}? it looks like show package just returns json

Reply:
show_package return the CKAN API response.
Here an example: https://github.com/datopian/ckan-ng-harvester-core/blob/3_avoid_409_error/tests_with_ckan/test_harvest.py#L78-L80
The response it's a dict. Here [an example](https://catalog.data.gov/api/3/action/package_show?id=8a311c18-3060-438e-a8c7-9e37bcde6529)